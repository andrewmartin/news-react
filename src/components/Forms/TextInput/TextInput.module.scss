@import "src/styles/variables";

.input {
  width: 100%;
  padding: 15px;
  transition: all .2s ease-out;
  border: 2px solid $gray;
  font-size: 16px;

  &:focus {
    border-color: $brand-primary;
    outline: none;
  }
}

.error {
  border-color: red !important;
}
