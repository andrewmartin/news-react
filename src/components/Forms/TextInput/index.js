import React, { Component } from 'react';
import cx from 'classnames';

import styles from './TextInput.module.scss';

export default class TextInput extends Component {
  state = {
    value: '',
    touched: false,
  };

  setValue = e => {
    const { onChange } = this.props;

    this.setState({
      value: e.target.value,
      touched: true,
    });
    onChange(e.target.value);
  };

  render() {
    const { className, onChange, required, ...rest } = this.props;
    const { value, touched } = this.state;

    return (
      <input
        className={cx(styles.input, className, {
          [styles.error]: required && touched && value === '',
        })}
        onChange={this.setValue}
        required={required}
        type="text"
        value={value}
        {...rest}
      />
    );
  }
}
