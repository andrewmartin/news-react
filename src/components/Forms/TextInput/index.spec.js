import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './index';

it('renders TextInput', () => {
  const tree = renderer.create(<TextInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
