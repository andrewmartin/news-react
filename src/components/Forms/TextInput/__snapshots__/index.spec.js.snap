// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextInput 1`] = `
<input
  className="input"
  onChange={[Function]}
  type="text"
  value=""
/>
`;
